src/py42/services/alerts.py

```diff
@@ -1,34 +1,34 @@
 import json
 
 from py42 import settings
-from py42._internal.compat import str
-from py42.clients import BaseClient
-from py42.clients.util import get_all_pages
+from py42._compat import str
 from py42.sdk.queries.query_filter import create_eq_filter_group
+from py42.services import BaseService
+from py42.services.util import get_all_pages
 
 
-class AlertClient(BaseClient):
+class AlertService(BaseService):
     _uri_prefix = u"/svc/api/v1/{0}"
 
     _CREATED_AT = u"CreatedAt"
     _RULE_METADATA = u"ruleMetadata"
 
-    def __init__(self, session, user_context):
-        super(AlertClient, self).__init__(session)
+    def __init__(self, connection, user_context):
+        super(AlertService, self).__init__(connection)
         self._user_context = user_context
 
     def search(self, query):
         query = self._add_tenant_id_if_missing(query)
         uri = self._uri_prefix.format(u"query-alerts")
-        return self._session.post(uri, data=query)
+        return self._connection.post(uri, data=query)
 
     def get_details(self, alert_ids):
         if not isinstance(alert_ids, (list, tuple)):
             alert_ids = [alert_ids]
         tenant_id = self._user_context.get_current_tenant_id()
         uri = self._uri_prefix.format(u"query-details")
         data = {u"tenantId": tenant_id, u"alertIds": alert_ids}
-        results = self._session.post(uri, data=json.dumps(data))
```

Comment:
What is going on here?  I don't see `json` as an explicit parameter for `Connection.post`

Reply:
I can make it explicit. I changed everything that was using `data=json.dumps()` to `json=dict` for a couple reasons:
1. There was already inconsistency here.
2. Because earlier versions of python are not guaranteed to have consistent outputs of `json.dumps()` for the same dict, it makes it difficult to test that a `post` was issued with the correct data. checking for dict equivalency is as simple as `assert dict1 == dict2`.